packages/flutter_android_lifecycle/pubspec.yaml

```diff
@@ -0,0 +1,60 @@
+name: flutter_android_lifecycle
+description: A new flutter plugin project.
+version: 0.0.1
```

Comment:
How do you think we should version this package? One thing we could do is try to be consistent with the Android lifecycle major version (e.g version this 2.0.0). Though if the Android lifecycle package gets a breaking change that does not affect plugins the versions will get out of sync...

Reply:
I generally like the idea of matching the version. You mentioned that there might be non-breaking changes for plugins, but what about the fact that `FlutterActivity` has a direct dependency on `Lifecycle` and `FlutterFragment` has an indirect dependency? Isn't it likely that a breaking change will require the embedding to upgrade, even if the plugins don't necessary care about that specific change? If so, then I think we can match versions. If not, then we might wanna look at alternatives.

Reply:
Is the scenario something like a lifecycle 3.0.0 which didn't change any API that can break plugins, but we bumped the dependency in the POM file to ask for lifecycle 3.0.0

I'd say that in that case we won't bump the major version of the `flutter_android_lifecycle` as clients that have used it's previous version need no change to their code...

Reply:
Lemme outline a scenario and see if I'm understanding correctly. Let's say...

Today:
embedding uses Lifecycle v1.1
plugins use Lifecycle v1.1

Tomorrow:
embedding uses Lifecycle v1.2 (with some kind of non-plugin related breaking change)
plugins use Lifecycle v1.1 (no change)

Is that the expectation? If so, will that even compile? CC @blasten

Reply:
There are 2 lifecycle packages involved: android.lifecycle (maven package) and flutter_android_lifecycle (pub package).

The plugins always use the same android.lifecycle as the embedding.
In the event of a breaking change to android.lifecycle that affects plugins we will bump the major version of flutter_android_lifecycle.

The major version number of flutter_android_lifecycle is only bumped when there's a breaking change to android.lifecycle that affects plugins.

One scenario I see is:

Time T:
embedding/plugins use android.lifecycle 2.2.2, plugins depend on flutter_android_lifecycle 2.2.2

Time T+1:
android.lifecycle 3.0.0 is release but does not require any changes in plugins so we do not bump the version of flutter_android_lifecycle, that means:
embedding/plugins use android.lifecycle 3.0.0, plugins depend on flutter_android_lifecycle 2.2.2

Reply:
We spoke offline. The concern I had shouldn't be a problem as long as plugins do not explicitly declare a lifecycle dependency, which we don't expect them to do.